src/mscorlib/corefx/System/Globalization/CalendarData.Unix.cs

```diff
@@ -128,28 +129,155 @@ private static bool GetCalendarInfo(string localeName, CalendarId calendarId, Ca
+                        // so read verbatim until the next single quote
+                        destination.Append(input[index++]);
+                        while (index < input.Length)
+                        {
+                            char current = input[index++];
+                            destination.Append(current);
+                            if (current == '\'')
+                            {
+                                break;
+                            }
+                        }
+                        break;
+                    case 'E':
+                    case 'e':
+                    case 'c':
+                        // 'E' in ICU is the day of the week, which maps to 3 or 4 'd's in .NET
+                        // 'e' in ICU is the local day of the week, which has no representation in .NET, but
+                        // maps closest to 3 or 4 'd's in .NET
+                        // 'c' in ICU is the stand-alone day of the week, which has no representation in .NET, but
+                        // maps closest to 3 or 4 'd's in .NET
+                        NormalizeDayOfWeek(input, destination, ref index);
+                        break;
+                    case 'L':
+                        // 'L' in ICU is the stand-alone name of the month,
+                        // which maps closest to 'M' in .NET since it doesn't support stand-alone month names in patterns
+                        int occurrences = CountOccurrences(input, 'L', ref index);
+                        if (occurrences > 4)
+                        {
+                            // 5 'L's in ICU is the super short name, which maps closest to MMM in .NET
+                            occurrences = 3;
+                        }
+
+                        destination.Append('M', occurrences);
+                        break;
+                    case 'G':
+                        // 'G' in ICU is the era, which maps to 'g' in .NET
+                        occurrences = CountOccurrences(input, 'G', ref index);
+
+                        // it doesn't matter how many 'G's, since .NET only supports 'g' or 'gg', and they
+                        // have the same meaning
+                        destination.Append('g');
+                        break;
+                    case 'y':
+                        // a single 'y' in ICU is the year with no padding or trimming.
+                        // a single 'y' in .NET is the year with 1 or 2 digits
+                        // so convert any single 'y' to 'yyyy'
+                        occurrences = CountOccurrences(input, 'y', ref index);
+                        if (occurrences == 1)
+                        {
+                            occurrences = 4;
+                        }
+                        destination.Append('y', occurrences);
+                        break;
+                    default:
+                        destination.Append(input[index++]);
```

Comment:
can we at least Assert on such cases? so we get some attention when someone introduce it?